build/uploadperftobenchview.sh

```diff
@@ -0,0 +1,115 @@
+#!/bin/bash
+
+# The intent of this script is upload produced performance results to BenchView in a CI context.
+#    There is no support for running this script in a dev environment.
+
+if ! env | grep -q ^perfWorkingDirectory=
+then
+    echo EnvVar perfWorkingDirectory should be set; exiting...
+    exit 1
+fi
+if ! env | grep -q ^configuration=
+then
+    echo EnvVar configuration should be set; exiting...
+    exit 1
+fi
+if ! env | grep -q ^architecture=
+then
+    echo EnvVar architecture should be set; exiting...
+    exit 1
+fi
+if ! env | grep -q ^OS=
+then
+    echo EnvVar OS should be set; exiting...
+    exit 1
+fi
+
+if ! env | grep -q ^runType=private
```

Comment:
This seems like the wrong way to check the values of environment variables.  I think the right way is something like this:

    if [ "$runType" = private ]; then

Reply:
done...